sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java

```diff
@@ -537,14 +1407,92 @@
 
         try {
             // If-Match == "*" to unconditionally update. This is in line with the existing client library behaviour.
-            return queuesClient.putWithResponseAsync(queue.getName(), createEntity, "*", withTracing)
+            return entityClient.putWithResponseAsync(queue.getName(), createEntity, "*", withTracing)
                 .onErrorMap(ServiceBusManagementAsyncClient::mapException)
                 .map(response -> deserializeQueue(response));
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
     }
 
+    /**
+     * Updates a subscription with its context.
+     *
+     * @param subscription Information about the subscription to update. You must provide all the property values
+     *     that are desired on the updated entity. Any values not provided are set to the service default values.
+     * @param context Context to pass into request.
+     *
+     * @return A Mono that completes with the updated {@link SubscriptionDescription}.
+     */
+    Mono<Response<SubscriptionDescription>> updateSubscriptionWithResponse(SubscriptionDescription subscription,
+        Context context) {
+        if (subscription == null) {
+            return monoError(logger, new NullPointerException("'subscription' cannot be null"));
+        }
+
+        final String topicName = subscription.getTopicName();
+        final String subscriptionName = subscription.getSubscriptionName();
+        if (topicName == null || topicName.isEmpty()) {
+            return monoError(logger, new IllegalArgumentException("'getTopicName' cannot be null or empty."));
```

Comment:
`topicName == null` : NullPointerException ? and add java doc for it.

Reply:
There is a null check when creating the actual subscription object. So this won't happen. I'll remove the double checks.